examples/sampling/graphbolt/lightning/node_classification.py

```diff
@@ -109,57 +134,50 @@ def configure_optimizers(self):
 
 
 class DataModule(LightningDataModule):
-    def __init__(self, fanouts, batch_size, num_workers):
+    def __init__(self, fanouts, batch_size, num_workers, labor):
         super().__init__()
         self.fanouts = fanouts
         self.batch_size = batch_size
         self.num_workers = num_workers
-        # TODO: Update with a publicly accessible URL once the dataset has been
-        # uploaded.
-        dataset = gb.OnDiskDataset(
-            "/home/ubuntu/workspace/example_ogbn_products/"
-        )
-        dataset.load()
+        dataset = gb.BuiltinDataset("ogbn-products").load()
         self.feature_store = dataset.feature
         self.graph = dataset.graph
         self.train_set = dataset.tasks[0].train_set
         self.valid_set = dataset.tasks[0].validation_set
         self.num_classes = dataset.tasks[0].metadata["num_classes"]
+        self.labor = labor
 
-    ########################################################################
-    # (HIGHLIGHT) The 'train_dataloader' and 'val_dataloader' hooks are
-    # essential components of the Lightning framework, defining how data is
-    # loaded during training and validation. In this example, we utilize a
-    # specialized 'graphbolt dataloader', which are concatenated by a series
-    # of datappipes, for these purposes.
-    ########################################################################
-    def train_dataloader(self):
+    def create_dataloader(self, node_set, is_train):
         datapipe = gb.ItemSampler(
-            self.train_set,
+            node_set,
             batch_size=self.batch_size,
-            shuffle=True,
-            drop_last=True,
+            shuffle=is_train,
+            drop_last=is_train,
```

Comment:
If `drop_last` is False, data maybe not evenly distributed across processes, my testing shows the entire procedure will be blocked, haven't you encounter this?

Reply:
I didn't try distributed training. I wasn't aware that changing drop_last would cause a hang like that. I can revert the change.

Reply:
So, should I pass `shuffle=True, drop_last=True` for validation like before?

Reply:
Yes, I think for the robustness, we need to set `drop_last=True`.